cgi-bin/DW/Media.pm

```diff
@@ -110,9 +110,18 @@ sub upload_media {
 
     # set the security
     my $sec = $opts{security} || 'public';
+    if ( $sec =~ /^(?:friends|access)$/ ) {
+        $sec = 'usemask' ;
+        $opts{allowmask} = 1;
+    }
     croak 'Invalid security for uploaded file.'
         unless $sec =~ /^(?:public|private|usemask)$/;
-    $opts{allowmask} = 0 unless defined $opts{allowmask} && $sec eq 'usemask';
+    if ( $sec eq 'usemask' ) {
+        warn "Must define allowmask when security is usemask"
+            if ! defined $opts{allowmask} && $LJ::IS_DEV_SERVER;
```

Comment:
The prior code would always set it to `0` if it was undefined, in this new code, it lets through `undef`. I think we need to set it to `0` explicitly if it's undefined.